src/eduid_webapp/email/views.py

```diff
@@ -219,28 +221,41 @@ def verify_link(user):
         db = current_app.proofing_statedb
         state = db.get_state_by_eppn_and_email(proofing_user.eppn, email, raise_on_missing=False)
 
+        url = urlparse.urljoin(current_app.config['DASHBOARD_URL'], 'emails')
+        url_parts = list(urlparse.urlparse(url))
+
         if state is None:
             current_app.logger.info("Missing state for verification code received for email {} "
                                     "and user {}.".format(email, user))
-            return redirect(current_app.config['SAML2_LOGIN_REDIRECT_URL'])
+            url_parts[4] = urlencode({'msg': ':ERROR:emails.unknown_email'})
+            url = urlparse.urlunparse(url_parts)
+            return redirect(url)
```

Comment:
Well not just the setting, as far as I can see, there is actually no part of the url that can be determined from the outside. Would you still check that the URL is correct? Since we've just built it from settings, I didn't think it was necessary...

Reply:
One thing that this opens is the podsibility of constructing URLs that will show an arbitrary msg to the user. For example, we could do something like:

https://dashboard.eduid.se/profile/personaldata?msg=foo

This will get you to the personal data form, with a message like "UNKNOWN MESSAGE ID (foo)".

Reply:
Maybe the unknown msg should not be shown here, to prevent showing the user a fraudulent message such as "UNKNOWN MESSAGE ID (hurry up and send all your bitcoins to me)"

Reply:
Yes that is a problem. I think that the best solution for this would be to never show messages with unkown message ids in production; so if we are in development mode (DEBUG=true), we show UNKNOWN MESSAGE ID (foo) for any unknown foo, but if we are in production, any unknown message id will be silently ignored.

This however must happen in eduid-html, so I'll implement it there.

Reply:
Well, perhaps not really silently ignored; if there is some unexpected 500, we may want to tell the user that there is a problem. So what I'd do, is, in development, show the message id as it is now, since it is quite useful to complete the collection of ranslatable messages; And, in production, only show a default and constant "there was an unexpected problem, please try again or contact the site administration".